checker/src/call_visitor.rs

```diff
@@ -1754,6 +1758,92 @@ impl<'call, 'block, 'analysis, 'compilation, 'tcx, E>
         self.use_entry_condition_as_exit_condition();
     }
 
+    #[logfn_inputs(TRACE)]
+    fn handle_write_bytes(&mut self) {
+        checked_assume!(self.actual_args.len() == 3);
+        let dest_path = Path::new_deref(self.actual_args[0].0.clone())
+            .refine_paths(&self.block_visitor.bv.current_environment);
+        let dest_type = self.actual_argument_types[0];
+        let source_path = self.actual_args[1].0.clone();
+        let byte_value = &self.actual_args[1].1;
+        let count_value = self.actual_args[2].1.clone();
+        let elem_type = self
+            .callee_generic_arguments
+            .expect("write_bytes<T>")
+            .get(0)
+            .expect("write_bytes<T>")
+            .expect_ty();
+        let mut elem_size = self.block_visitor.bv.type_visitor.get_type_size(elem_type);
+        fn repeated_bytes(mut elem_size: u64, byte_value: &Rc<AbstractValue>) -> Rc<AbstractValue> {
+            let const_8: Rc<AbstractValue> = Rc::new(8u128.into());
+            let mut source_value = byte_value.clone();
+            while elem_size > 1 {
+                source_value = source_value
+                    .shift_left(const_8.clone())
+                    .bit_or(byte_value.clone());
+                elem_size -= 1;
+            }
+            source_value
+        }
+        if elem_type.is_primitive() {
+            let dest_pattern = Path::new_slice(dest_path, count_value);
+            let source_value = repeated_bytes(elem_size, byte_value);
+            self.block_visitor
+                .bv
+                .current_environment
+                .update_value_at(dest_pattern, source_value);
```

Comment:
Is it supposed to be a `copy_or_move_elements` call, because the target path is a pattern?

Reply:
No, we don't want expansion of the pattern at this point since the caller is going to do that anyway.

Reply:
Ah, the caller will do that during refinement?

Reply:
The caller's caller, yes.